Mathlib/Computability/TuringDegree.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2025 Tanner Duve. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tanner Duve, Elan Roth
+-/
+import Mathlib.Computability.Partrec
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Oracle Computability and Turing Degrees
+
+This file defines a model of oracle computability, introduces Turing reducibility and equivalence,
+proves that Turing equivalence is an equivalence relation, and defines Turing degrees as the
+quotient under this relation.
```

Comment:
I think it's helpful to say in the module docstring how this relates to partial recursive functions too